src/chat/ChatScreen.js

```diff
@@ -139,6 +141,8 @@ export default function ChatScreen(props: Props) {
           narrow={narrow}
           editMessage={editMessage}
           completeEditMessage={() => setEditMessage(null)}
+          defaultTopic={editMessage ? editMessage.topic : undefined}
+          defaultMessage={editMessage ? editMessage.content : draft}
```

Comment:
> I'm having some trouble verifying this commit as a pure refactor, and I think a main reason is that I'm not sure why `editMessage` is making an appearance in it, and I don't find it explained.

Hmmm, this might be an artifact of the way that I split up these commits. (Or maybe it's needed, I'm not actually sure).

It's basically because of the code in `UNSAFE_componentWillReceiveProps`, which is `editMessage`-dependent. Perhaps that doesn't matter for this commit, and should only be added once I remove `UNSAFE_componentWillReceiveProps`, though?

It's been sort of frustrating for me to try to get this PR into a state with a bunch of atomic commits, since the fundamental thing that I'm trying to fix is that `ComposeBox` is sort of a mess where changing one thing has a bunch of cascading effects most of the time.

I can also appreciate that it's much easier to review things that are small atomic commits, so I'm happy to work towards that, I'm just not confident in my ability to figure out exactly what each commit does.

I do think you can probably convince yourself that this commit is NFC, though — if `editMessage` is non-null, that means that `UNSAFE_componentWillReceiveProps` will run, and so none of what we set in the lines that are changed will matter. I guess that means that I should do that as part of a separate commit, I'll try to fix that up...

Reply:
> I'm also wondering why, in the case where `editMessage` is `null`, is there the following asymmetry between `defaultTopic` and `defaultMessage`: The value passed as `defaultTopic` is an emptiness value (`undefined`), leaving `ComposeBox` to fill in a meaningful value, while the value passed as `defaultMessage` is meaningful (`draft`).

This isn't quite correct, since `draft` can be `undefined` — it's possible that both will be unset, or it's possible that just message will be set.

The reason for doing it this way is that we don't save drafts of topic names (except sort of incidentally, in the fact that we store them by narrow). I'm not really sure what the best path forward here is — I think in involves rethinking how drafts work in general.